src/pyodide/internal/python.js

```diff
@@ -2,6 +2,23 @@ import { parseTarInfo } from "pyodide-internal:tar";
 import { createTarFS } from "pyodide-internal:tarfs";
 import { createMetadataFS } from "pyodide-internal:metadatafs";
 import { default as ArtifactBundler } from "pyodide-internal:artifacts";
+import { default as LOCKFILE } from "pyodide-internal:generated/pyodide-lock.json";
+
+const canonicalizeNameRegex = /[-_.]+/g;
+
+/**
+ * Normalize a package name. Port of Python's packaging.utils.canonicalize_name.
+ * @param name The package name to normalize.
+ * @returns The normalized package name.
+ * @private
+ */
+export function canonicalizePackageName(name) {
+  return name.replace(canonicalizeNameRegex, "-").toLowerCase();
+}
```

Comment:
I still don't think you need to do this, Pyodide is already doing it:
https://github.com/pyodide/pyodide/blob/main/src/js/load-package.ts#L150

Reply:
We need the names to be normalized on edgeworker since we're not calling loadPackage and the names are normalized in the tar file.

Reply:
We are calling `recursiveDependencies` which should handle this. But it's fine for now, we should fix it upstream and remove later but it's pretty low priority.